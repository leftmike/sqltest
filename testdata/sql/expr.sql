--
-- Test expressions
--

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (c1 int, c2 int);

INSERT INTO tbl1 VALUES
    (1, 10),
    (2, 20),
    (3, 30),
    (4, 40),
    (5, 50),
    (6, 60),
    (7, 70),
    (8, 80);

DROP TABLE IF EXISTS tbl2;

CREATE TABLE tbl2 (c1 text, c2 int);

INSERT INTO tbl2 VALUES
    ('one', 1),
    ('two', 2),
    ('three', 3),
    ('four', 4),
    ('five', 5),
    ('six', 6),
    ('seven', 7),
    ('eight', 8);

SELECT (SELECT c1 FROM tbl2 WHERE c2 = tbl1.c1) AS e1, c2 FROM tbl1;

SELECT (SELECT c1 FROM tbl2 WHERE c2 = tbl1.c1 AND c1 != 'one') AS e1, c2 FROM tbl1;

SELECT (SELECT * FROM (SELECT c1 FROM tbl2 WHERE c2 = tbl1.c1) AS tbl3) AS e1, c2 FROM tbl1;
