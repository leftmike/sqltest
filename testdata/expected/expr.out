--
-- Test expressions
--
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (c1 int, c2 int);
INSERT INTO tbl1 VALUES
    (1, 10),
    (2, 20),
    (3, 30),
    (4, 40),
    (5, 50),
    (6, 60),
    (7, 70),
    (8, 80);
DROP TABLE IF EXISTS tbl2;
CREATE TABLE tbl2 (c1 text, c2 int);
INSERT INTO tbl2 VALUES
    ('one', 1),
    ('two', 2),
    ('three', 3),
    ('four', 4),
    ('five', 5),
    ('six', 6),
    ('seven', 7),
    ('eight', 8);
SELECT (SELECT c1 FROM tbl2 WHERE c2 = tbl1.c1) AS e1, c2 FROM tbl1;
      e1 c2
      -- --
 1 eight 80
 2  five 50
 3  four 40
 4   one 10
 5 seven 70
 6   six 60
 7 three 30
 8   two 20
(8 rows)
SELECT (SELECT c1 FROM tbl2 WHERE c2 = tbl1.c1 AND c1 != 'one') AS e1, c2 FROM tbl1;
      e1 c2
      -- --
 1       10
 2 eight 80
 3  five 50
 4  four 40
 5 seven 70
 6   six 60
 7 three 30
 8   two 20
(8 rows)
SELECT (SELECT * FROM (SELECT c1 FROM tbl2 WHERE c2 = tbl1.c1) AS tbl3) AS e1, c2 FROM tbl1;
      e1 c2
      -- --
 1 eight 80
 2  five 50
 3  four 40
 4   one 10
 5 seven 70
 6   six 60
 7 three 30
 8   two 20
(8 rows)
